bertin.md

```diff
@@ -0,0 +1,501 @@
+
+- In order to get home, I have to **(Spain's word for) drive** my (Spain's word for) car.  
+  (Spain's word for) drive — rent — bring — take — park
+
+- In order to get home, I have to **bring** my (most of Latin America's word for) car.  
+  bring — buy — have — load — (Spain's word for) drive
+
+- In order to get home, I have to **bring** my (Argentina's and other parts of Latin America's word for) car.  
+  bring — have — (Spain's word for) drive — take — load  
+
+## Analysis
+
+The performance of our BERTIN models has been, in general, very good. Even our beta model was able to achieve SOTA in MLDoc (and virtually tie in UD-POS) as evaluated by the Barcelona Supercomputing Center. In the main masked-language task our models reach values between 0.65 and 0.69, which foretells good results for downstream tasks.
+
+It should be noted, that it is certainly possible that any of the models —ours or otherwise— could be carefully tuned to achieve better results at a given task, and it is a possibility that the best tuning might result in a new "winner" for that category. What we can claim is that, under standard training conditions, our models are remarkably performant. In particular, `Gaussian` perplexity sampling seems to produce consistently solid models, taking the lead in four of the seven tasks analysed.
+
+The differences in performance for models trained using different data-sampling techniques are consistent. `Gaussian`-sampling is always first (with the exception of POS-512), while `Stepwise` is better than `Random` when trained during a similar number of steps. This proves that the sampling technique is, indeed, relevant and a more thorough statistical analysis could help optimize these sampling strategies.
+
+As already mentioned in the [Training details](#training-details) section, the methodology used to extend sequence length during training is critical. The `Random`-sampling model took an important hit in performance in this process, while `Gaussian`-512 ended up with better metrics than `Gaussian`-128, in both the main masked-language task and the downstream datasets. The key difference was that `Random` kept the optimizer learning rate scheduler intact while `Gaussian` re-started the schedule. This difference is likely related to the fact that the learning rate was very low close to the end of training, and the adjustments needed after a change in sequence length require a larger learning rate. We believe this is an important topic of research, but our preliminary data suggests that restarting the learning rate scheduler is a safe alternative when in doubt or if computational resources are scarce.
+
+# Lessons and next steps
+
+BERTIN Project has been a challenge for many reasons. Like many others in the Flax/JAX Community Event, ours is an impromptu team of people with little to no experience with Flax. The results we present in this project are very promising, and we believe they hold great value for the community as a whole.
```

Comment:
```suggestion
BERTIN Project has been a challenge for many reasons. Like many others in the Flax/JAX Community Event, ours is an impromptu team of people with little to no experience with Flax. Moreover, other than main goal of producing a RoBERTa model, nothing was decided upon the start of the project, so we had to spend some time organizing how we would work and communicate, and debating and agreeing on the approaches we would like to take given the limitations in time and resources, since most of us also had full-time jobs to attend to. Notwithstanding, the results we present in this project are very promising, and we believe they hold great value for the community as a whole.
```